notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dd91d16b-1880-4a4c-8e8a-650075e26015",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "REPLACE INTO \"wikipedia_events_3\" OVERWRITE ALL\n",
+    "WITH \"ext\" AS \n",
+    "(\n",
+    "    SELECT *\n",
+    "    FROM TABLE(\n",
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\n",
+    "          \"uris\":[ \"https://druid.apache.org/data/wikipedia.json.gz\",\n",
+    "                   \"https://druid.apache.org/data/wikipedia.json.gz\",\n",
+    "                   \"https://druid.apache.org/data/wikipedia.json.gz\"\n",
+    "                 ]\n",
+    "         }',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    ") EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR, \"flags\" VARCHAR, \"isUnpatrolled\" VARCHAR, \"page\" VARCHAR, \"diffUrl\" VARCHAR, \"added\" BIGINT, \"comment\" VARCHAR, \"commentLength\" BIGINT, \"isNew\" VARCHAR, \"isMinor\" VARCHAR, \"delta\" BIGINT, \"isAnonymous\" VARCHAR, \"user\" VARCHAR, \"deltaBucket\" BIGINT, \"deleted\" BIGINT, \"namespace\" VARCHAR, \"cityName\" VARCHAR, \"countryName\" VARCHAR, \"regionIsoCode\" VARCHAR, \"metroCode\" BIGINT, \"countryIsoCode\" VARCHAR, \"regionName\" VARCHAR))\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "druid.display.run_task(sql)\n",
+    "druid.sql.wait_until_ready('wikipedia_events_3')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41ce84c5-2c1c-4c6a-b9f4-7a22e2216ec5",
+   "metadata": {},
+   "source": [
+    "Let's look at the data now. The quantities are 3 times larger than before because we loaded the same file three times:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f278c0d2-9fd5-4a7b-bb65-a6db7cf98971",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.display.sql(\"\"\"\n",
+    "SELECT channel, count(*) num_events\n",
+    "FROM \"wikipedia_events_3\" \n",
+    "GROUP BY 1 \n",
+    "ORDER BY 2 DESC \n",
+    "LIMIT 10\n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "751316a2-bd0c-43eb-a74f-fed15782f54d",
+   "metadata": {},
+   "source": [
+    "#### Context Parameters\n",
+    "Certain aspects of the ingestion can be controlled using [context parameter](https://druid.apache.org/docs/latest/multi-stage-query/reference.html#context-parameters)s. This section discussed two of the commonly used parameters:\n",
+    "\n",
+    "##### maxNumTasks\n",
+    "The Multi-stage Query Framework uses parallel workers to execute each stage of the ingestion process. Each stage creates output partitions that organize the data in preparation for the next stage. \n",
```

Comment:
Rather than an explanation (which is in the docs) is there a way that you can demonstrate this (and the other paragraphs below)?

Reply:
Not sure what you mean by demonstrating.... the resulting segments view is how I am demonstrating the use of the parameters. We would have to change the display.run_task to show more about the workers running...let's do that separately.